tests/test_pandas.py

```diff
@@ -142,71 +145,60 @@ def get_engine_with_numpy(db_parameters, user=None, password=None, account=None)
 
 def test_numpy_datatypes(db_parameters):
     engine = get_engine_with_numpy(db_parameters)
-    conn = engine.connect()
-    try:
-        specific_date = np.datetime64("2016-03-04T12:03:05.123456789")
-        with conn.begin():
-            conn.exec_driver_sql(
-                "CREATE OR REPLACE TABLE {name}("
-                "c1 timestamp_ntz)".format(name=db_parameters["name"])
-            )
-        with conn.begin():
-            conn.exec_driver_sql(
-                "INSERT INTO {name}(c1) values(%s)".format(name=db_parameters["name"]),
-                (specific_date,),
-            )
-        with conn.begin():
-            df = pd.read_sql_query(
-                text("SELECT * FROM {name}".format(name=db_parameters["name"])), conn
-            )
-            assert df.c1.values[0] == specific_date
-    finally:
-        conn.exec_driver_sql(
-            "DROP TABLE IF EXISTS {name}".format(name=db_parameters["name"])
-        )
-        conn.close()
-        engine.dispose()
+    with engine.connect() as conn:
+        try:
+            specific_date = np.datetime64("2016-03-04T12:03:05.123456789")
+            with conn.begin():
+                conn.exec_driver_sql(
+                    f"CREATE OR REPLACE TABLE {db_parameters['name']}(c1 timestamp_ntz)"
+                )
+                conn.exec_driver_sql(
+                    f"INSERT INTO {db_parameters['name']}(c1) values(%s)",
+                    (specific_date,),
+                )
+                df = pd.read_sql_query(
+                    text(f"SELECT * FROM {db_parameters['name']}"), conn
+                )
+                assert df.c1.values[0] == specific_date
+        finally:
+            conn.exec_driver_sql(f"DROP TABLE IF EXISTS {db_parameters['name']}")
+            engine.dispose()
 
 
 def test_to_sql(db_parameters):
     engine = get_engine_with_numpy(db_parameters)
-    conn = engine.connect()
-    total_rows = 10000
-    conn.exec_driver_sql(
-        """
-create or replace table src(c1 float)
- as select random(123) from table(generator(timelimit=>1))
- limit {}
-""".format(
-            total_rows
+    with engine.connect() as conn:
+        total_rows = 10000
+        conn.exec_driver_sql(
+            textwrap.dedent(
+                f"""\
+            create or replace table src(c1 float)
+            as select random(123) from table(generator(timelimit=>1))
+            limit {total_rows}
+            """
```

Comment:
Looks a little weird
```suggestion
                f"""\
                create or replace table src(c1 float)
                as select random(123) from table(generator(timelimit=>1))
                limit {total_rows}
                """
```